lib/ansible/plugins/inventory/openstack.py

```diff
@@ -0,0 +1,308 @@
+    options:
+        show_all:
+            description: toggles showing all vms vs only those with a working IP
+            type: boolean
+            default: False
+        use_server_id:
+            description: |
+                Register every host in the inventory with its server id UUID
+                instead of the server name and make a group for each server
+                name. The default behavior is to register each host in the
+                inventory by name and only register by UUID if there is more
+                than one instance of a given name. (Note, the default value of
+                this is opposite from the old openstack.py inventory script's
+                option use_hostnames)
+            type: boolean
+            default: False
+        expand_hostvars:
+            description: |
+                Run extra commands on each host to fill in additional
+                information about the host. May interrogate cinder and
+                neutron and can be expensive for people with many hosts.
+                (Note, the default value of this is opposite from the default
+                old openstack.py inventory script's option expand_hostvars)
+            type: boolean
+            default: False
+        private:
+            description: |
+                Use the private interface of each server, if it has one, as
+                the host's IP in the inventory. This can be useful if you are
+                running ansible inside a server in the cloud and would rather
+                communicate to your servers over the private network.
+            type: boolean
+            default: False
+        only_clouds:
+            description: |
+                List of clouds from clouds.yaml to use, instead of using
+                the whole list.
+            type: list
+            default: []
+        fail_on_errors:
+            description: |
+                Causes the inventory to fail and return no hosts if one cloud
+                has failed (for example, bad credentials or being offline).
+                When set to False, the inventory will return as many hosts as
+                it can from as many clouds as it can contact. (Note, the
+                default value of this is opposite from the old openstack.py
+                inventory script's option fail_on_errors)
+            type: boolean
+            default: False
+        clouds_yaml_path:
+            description: |
+                Override path to clouds.yaml file. If this value is given it
+                will be searched first. The default path for the
+                ansible inventory adds /etc/ansible/openstack.yaml and
+                /etc/ansible/openstack.yml to the regular locations documented
+                at https://docs.openstack.org/os-client-config/latest/user/configuration.html#config-files
+            type: string
+            default: None
```

Comment:
possibly add compose/groups for constructed vars and groups (see virtualbox for example)

Reply:
Thanks, I added those. I'm also going to make another pass later and use that facility for the constructed groups we already make. Super helpful.